dokka-hide-internal/src/main/kotlin/com/revenuecat/dokka/plugin/hideinternal/HideInternalRevenueCatAPIPlugin.kt

```diff
@@ -0,0 +1,40 @@
+package com.revenuecat.dokka.plugin.hideinternal
+
+import org.jetbrains.dokka.base.DokkaBase
+import org.jetbrains.dokka.base.transformers.documentables.SuppressedByConditionDocumentableFilterTransformer
+import org.jetbrains.dokka.model.Annotations
+import org.jetbrains.dokka.model.Documentable
+import org.jetbrains.dokka.model.properties.WithExtraProperties
+import org.jetbrains.dokka.plugability.DokkaContext
+import org.jetbrains.dokka.plugability.DokkaPlugin
+import org.jetbrains.dokka.plugability.DokkaPluginApiPreview
+import org.jetbrains.dokka.plugability.PluginApiPreviewAcknowledgement
+
+class HideInternalRevenueCatAPIPlugin : DokkaPlugin() {
```

Comment:
I guess this has to be public?

Reply:
I think so yes, but it's not published to Maven (unless I'm overlooking something). It's just used during docs generation, which happens locally ("locally" on CI).

Reply:
Did some experiments, and it has to be public indeed. It's using Java's ServiceLoader mechanism to discover the plugin class, and it can't find it if it's private. Internal works, but that's probably because ServiceLoaders are a Java thing, and internal in Kotlin = public in Java.